src/utils/styleDomMapper.js

```diff
@@ -0,0 +1,127 @@
+var List = require('csso:utils/list.js');
+var translate = require('csso:utils/translate.js');
+var walk = require('utils.walker');
+
+module.exports = function StyleDOMMapper() {
+  var selectorElementMap;
+  var elementSelectorMap;
+  var mapper = {
+    bySelector: function(token) {
+      return selectorElementMap.get(token);
+    },
+    byElement: function(element) {
+      return elementSelectorMap.get(element);
+    },
+    removeSelector: function(selector) {
+      var mappedElements = selectorElementMap.get(selector);
+
+      if (mappedElements) {
+        for (var i = 0; i < mappedElements.length; i++) {
+          var mappedSelectors = elementSelectorMap.get(mappedElements[i]);
+          var ix = mappedSelectors.indexOf(selector);
+
+          if (ix > -1) {
+            mappedSelectors.splice(ix, 1);
+
+            if (!mappedSelectors.length) {
+              elementSelectorMap.delete(mappedElements[i]);
+            }
+          }
+        }
+
+        selectorElementMap.delete(selector);
+      }
+    },
+    removeElement: function(element) {
+      var mappedSelectors = elementSelectorMap.get(element);
+
+      if (mappedSelectors) {
+        for (var i = 0; i < mappedSelectors.length; i++) {
+          var mappedElements = selectorElementMap.get(mappedSelectors[i]);
+          var ix = mappedElements.indexOf(element);
+
+          if (ix > -1) {
+            mappedElements.splice(ix, 1);
+
+            if (!mappedElements.length) {
+              selectorElementMap.delete(mappedSelectors[i]);
+            }
+          }
+        }
+
+        elementSelectorMap.delete(element);
+      }
+    },
+    linkElements: function(token, elements) {
+      var selector = translate(token);
+      var mappedElements = selectorElementMap.get(token) || [];
+
+      elements = basis.array.from(elements);
```

Comment:
`elements` may be not an array
`basis.array.from` is used for insurance that `elements` will be array
although, we can do something like `Array.isArray(elements) ? element : basis.array.from(elements)`
thanks